### PR TITLE
[FEATURE] Search projects via full-text search

### PR DESCRIPTION
**Pull request by volkanunsal**
This feature allows the API to accept full text search queries to the endpoint `/api/projects` using query parameter `q`. It indexes `name`, `categories` and `description` fields. It also allows use to get the results only as ids by passing the `only_ids` parameter.

### CODE REVIEW

**Comment by volkanunsal** [comment]
Travis is complaining that `cfapi` database is not found. Looks like a configuration error somewhere...

File: app.py [hunk 1]
```diff
@@ -534,7 +563,11 @@ def paged_results(query, page, per_page, querystring=''):
     '''
     total = query.count()
     last, offset = page_info(query, page, per_page)
-    model_dicts = [o.asdict(True) for o in query.limit(per_page).offset(offset)]
+    if(querystring.find("only_ids") != -1):
+        model_dicts = [o.id for o in query.limit(per_page).offset(offset)]
+    else:
+        model_dicts = [o.asdict(True) for o in query.limit(per_page).offset(offset)]
```
Comment (volkanunsal) [inline app.py]: Does anyone know how to exclude an attribute from the API output here? I've tried `options(defer('tsv_body'))` already. That doesn't work with `asdict(True)`

**Comment by ondrae** [comment]
@volkanunsal This is great work. I didn't know about the tsvector approach yet it makes a lot of sense.

Can you tell me more about the `sort_by` params you added on the project search?

**Comment by volkanunsal** [comment]
`sort_by` allows you to tell the API which criteria to use to sort the results. For example, you can sort by `relevance` or by `last_updated`.

**Comment by volkanunsal** [comment]
It might be good to add another option to sort by `most_starred`.

**Comment by ondrae** [comment]
I've got both of these PR's combined up on [our staging server](http://cfapi-staging.herokuapp.com/api/) to check out how they work with a full database. I've started adding [a little bit more code too](https://github.com/codeforamerica/cfapi/tree/staging), I'll add some documentation next.

**Comment by ondrae** [comment]
You started out by creating a search endpoint instead of parameter. Why the change? I don't have an opinion either way.

**Comment by volkanunsal** [comment]
I changed it because the resource endpoints were already doing some search and filtering already, and what I was adding was just another type of search, so I didn't see a reason to introduce new vocabulary into the domain.

**Comment by volkanunsal** [comment]
I hope somebody can suggest a more permanent solution to removing the `tsv_body` field from the API response. My ad hoc solution is only removing it from top-level api responses, e.g. from `/projects` but not from `/issues`, for instance. because the projects are nested under `project` field in that case.

**Comment by ondrae** [comment]
I think I've caught that in the asdict() function of orgs and projects. See https://github.com/codeforamerica/cfapi/blob/staging/app.py#L223

**Comment by volkanunsal** [comment]
Awesome!

**Comment by ondrae** [comment]
Merged! Thanks Volkan. I've closed #194. 

I also added in the `github_details` to be indexed on projects. This adds a bunch of junk to that tsv search field, but it helps me get the project language from the json text. Know anyway to just add that one field of a json blob to be included?

**Pull request merged**